algorithms/src/snark/groth16/generator.rs

```diff
@@ -130,6 +139,19 @@ impl<E: PairingEngine> ConstraintSystem<E::Fr> for KeypairAssembly<E> {
     fn num_private_variables(&self) -> usize {
         self.num_private_variables
     }
+
+    fn optimization_goal(&self) -> OptimizationGoal {
+        self.optimization_goal.clone()
```

Comment:
since `OptimizationGoal` derives `Copy,` this clone (and all other clones of `OptimizationGoal` in this PR) shouldn't be necessary

Reply:
I now noticed `clippy` already complained about this later on; weird that it didn't notice this here